src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1593,6 +1593,21 @@ private static PropertySymbol GetPropertySymbol(BoundExpression expr, out BoundE
             return propertySymbol;
         }
 
+        internal static Symbol? GetIndexerSymbol(BoundExpression? e)
+        {
+            return e switch
+            {
+                null => null,
+                BoundIndexOrRangePatternIndexerAccess implicitIndexerAccess => GetIndexerSymbol(implicitIndexerAccess.IndexerAccess),
+                BoundIndexerAccess indexerAccess => indexerAccess.Indexer,
+                BoundCall call => call.Method,
+                BoundArrayAccess arrayAccess => arrayAccess.ExpressionSymbol,
```

Comment:
>arrayAccess.ExpressionSymbol [](http://example.com/codeflow?start=48&length=28)

What is this going to return?

Reply:
Changing the array case to just return null.